repositories/src/test/scala/cromwell/pipeline/datastorage/dao/repository/utils/TestProjectConfigurationUtils.scala

```diff
@@ -0,0 +1,51 @@
+package cromwell.pipeline.datastorage.dao.repository.utils
+
+import java.nio.file.{ Path, Paths }
+
+import cromwell.pipeline.datastorage.dto._
+import org.scalacheck.Gen
+
+object TestProjectConfigurationUtils {
+  lazy val typedValueGen: Gen[TypedValue] =
+    Gen.oneOf(stringTypedGen, fileTypedGen, intTypedGen, floatTypedGen, booleanTypedGen)
+
+  lazy val fileParameterGen: Gen[FileParameter] =
+    for {
+      nameField <- stringGen
+      valueField <- typedValueGen
+    } yield FileParameter(nameField, valueField)
+
+  lazy val projectFileConfigurationGen: Gen[ProjectFileConfiguration] =
+    for {
+      pathField <- fileGen
+      inputsField <- Gen.nonEmptyListOf(fileParameterGen).suchThat(_.length < 4)
+    } yield ProjectFileConfiguration(pathField, inputsField)
+
+  lazy val projectConfigurationGen: Gen[ProjectConfiguration] =
+    for {
+      projectIdField <- projectIDGen
+      configurationsField <- Gen.nonEmptyListOf(projectFileConfigurationGen).suchThat(_.length < 4)
+    } yield ProjectConfiguration(projectIdField, configurationsField)
+
+  private lazy val stringGen: Gen[String] = Gen.alphaLowerStr.suchThat(str => str.length < 128 && str.nonEmpty)
```

Comment:
It's better to change `suchThat` onto `repeatUntil` because right now this gen can return None if it will be a string with higher length

Reply:
Maybe you mean retryUntil? Because it doesn't exist

Reply:
Yep)